outputs.tf

```diff
@@ -1,14 +1,29 @@
 output "repository" {
-  description = "repository resource"
+  description = "All outputs of the repository."
   value       = try(aws_ecr_repository.repository[0], null)
 }
 
+output "repository_arn" {
+  description = "The ARN of the repository."
+  value       = try(aws_ecr_repository.repository[0].arn, null)
+}
+
+output "repository_id" {
+  description = "The Id of the repository."
+  value       = try(aws_ecr_repository.repository[0].id, null)
+}
+
+output "repository_url" {
+  description = "The URL of the repository."
+  value       = try(aws_ecr_repository.repository[0].repository_url, null)
+}
+
```

Comment:
why? you can get them from the repository itself
```
module.name.repository.arn
module.name.repository_arn
```

Reply:
fixed in aaa3cc7

Reply:
My take on outputs:

I don't have a strong opinion yet but here are some bullet points that lead me in the direction of explicit outputs:

- returning the whole object doesn't really create a contract / valid API. What if the internal representation changes due to Provider upgrade?  The module won't be able to guarantee backward compatibility anymore. If we return explicit outputs we can guarantee BC.
- What does the user really want? Returning the whole resource will most likely end in a scenario in which the user only uses 20% of the outputs anyways ... do we really need all information?
- We might return sensitive outputs per default if we always return the whole object?
- No convenient API - the user would need to check the terraform providers documentation to get an overview of the outputs.

I will post this in the related [trello task](https://trello.com/c/P4D2Y0BC/4-arrange-a-unified-output-format-for-all-terraform-modules) also. 
wdyt?

Also: I think we might be ready to get this merged now ? :)